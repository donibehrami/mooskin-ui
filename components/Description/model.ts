import { IBoxComponentProps } from '../Box/model';

export interface IDescriptionComponentProps extends IBoxComponentProps {
    /** Description children */
    children: string;
}
