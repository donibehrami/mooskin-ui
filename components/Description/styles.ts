import styled from 'styled-components';

import Box from '../Box/Box';

import variables from '../_utils/globals/variables';

import { IDescriptionComponentProps } from './model';

export const StyledDescription = styled(Box)<IDescriptionComponentProps>`
    margin-top: 10px;
    font-family: Hind;
    font-size: 12px;
    color: ${(props) => props.theme.inputFont || variables.inputFont};
    width: fit-content;
`;
