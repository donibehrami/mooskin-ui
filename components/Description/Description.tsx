import * as React from 'react';

import { IDescriptionComponentProps } from './model';
import { StyledDescription } from './styles';

export const Description: React.FC<IDescriptionComponentProps> = (props) => {
    return <StyledDescription {...props} />;
};

Description.defaultProps = {
    className: '',
    style: {}
};

Description.displayName = 'Description';

export default Description;
