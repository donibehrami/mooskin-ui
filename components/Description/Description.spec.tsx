import * as React from 'react';
import Description from './Description';

import { shallow } from 'enzyme';

describe('Description', () => {
    test('renders correctly', () => {
        const tree = shallow(<Description>Description here!</Description>);
        expect(tree).toMatchSnapshot();
    });
});
