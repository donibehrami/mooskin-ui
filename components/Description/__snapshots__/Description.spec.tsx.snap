// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description renders correctly 1`] = `
<Description
  className=""
  style={Object {}}
>
  <Component
    className=""
    palette={
      Object {
        "backgroundColors": Object {
          "button": "#3fbaca",
          "buttonDisabled": "#e2e2e2",
          "common": "#ffffff",
          "main": "#f7f7f7",
          "secondaryToggle": "#ffffff",
          "switchDisabled": "#e7e7e7",
          "switchHandle": "#ffffff",
          "switchOff": "#9d9d9d",
          "switchOn": "#3fbaca",
          "tag": "#293346",
          "toggle": "#425579",
        },
        "borderColors": Object {
          "common": "#e2e2e2",
        },
        "fontColors": Object {
          "anchor": "#3fbaca",
          "button": "#ffffff",
          "common": "#293346",
          "description": "#9d9d9d",
          "disabledAnchor": "#9d9d9d",
          "label": "#2d2d2d",
          "secondaryToggle": "#9d9d9d",
          "tableHeader": "#425579",
          "tag": "#ffffff",
          "text": "#2d2d2d",
          "toggle": "#ffffff",
        },
      }
    }
    style={Object {}}
  >
    <StyledDescription
      className=""
      palette={
        Object {
          "backgroundColors": Object {
            "button": "#3fbaca",
            "buttonDisabled": "#e2e2e2",
            "common": "#ffffff",
            "main": "#f7f7f7",
            "secondaryToggle": "#ffffff",
            "switchDisabled": "#e7e7e7",
            "switchHandle": "#ffffff",
            "switchOff": "#9d9d9d",
            "switchOn": "#3fbaca",
            "tag": "#293346",
            "toggle": "#425579",
          },
          "borderColors": Object {
            "common": "#e2e2e2",
          },
          "fontColors": Object {
            "anchor": "#3fbaca",
            "button": "#ffffff",
            "common": "#293346",
            "description": "#9d9d9d",
            "disabledAnchor": "#9d9d9d",
            "label": "#2d2d2d",
            "secondaryToggle": "#9d9d9d",
            "tableHeader": "#425579",
            "tag": "#ffffff",
            "text": "#2d2d2d",
            "toggle": "#ffffff",
          },
        }
      }
      style={Object {}}
    >
      <Box
        className="sc-gtssRu dZcusW"
        palette={
          Object {
            "backgroundColors": Object {
              "button": "#3fbaca",
              "buttonDisabled": "#e2e2e2",
              "common": "#ffffff",
              "main": "#f7f7f7",
              "secondaryToggle": "#ffffff",
              "switchDisabled": "#e7e7e7",
              "switchHandle": "#ffffff",
              "switchOff": "#9d9d9d",
              "switchOn": "#3fbaca",
              "tag": "#293346",
              "toggle": "#425579",
            },
            "borderColors": Object {
              "common": "#e2e2e2",
            },
            "fontColors": Object {
              "anchor": "#3fbaca",
              "button": "#ffffff",
              "common": "#293346",
              "description": "#9d9d9d",
              "disabledAnchor": "#9d9d9d",
              "label": "#2d2d2d",
              "secondaryToggle": "#9d9d9d",
              "tableHeader": "#425579",
              "tag": "#ffffff",
              "text": "#2d2d2d",
              "toggle": "#ffffff",
            },
          }
        }
        style={Object {}}
      >
        <Component
          className="sc-gtssRu dZcusW"
          palette={
            Object {
              "backgroundColors": Object {
                "button": "#3fbaca",
                "buttonDisabled": "#e2e2e2",
                "common": "#ffffff",
                "main": "#f7f7f7",
                "secondaryToggle": "#ffffff",
                "switchDisabled": "#e7e7e7",
                "switchHandle": "#ffffff",
                "switchOff": "#9d9d9d",
                "switchOn": "#3fbaca",
                "tag": "#293346",
                "toggle": "#425579",
              },
              "borderColors": Object {
                "common": "#e2e2e2",
              },
              "fontColors": Object {
                "anchor": "#3fbaca",
                "button": "#ffffff",
                "common": "#293346",
                "description": "#9d9d9d",
                "disabledAnchor": "#9d9d9d",
                "label": "#2d2d2d",
                "secondaryToggle": "#9d9d9d",
                "tableHeader": "#425579",
                "tag": "#ffffff",
                "text": "#2d2d2d",
                "toggle": "#ffffff",
              },
            }
          }
          style={Object {}}
        >
          <StyledBox
            className="sc-gtssRu dZcusW"
            palette={
              Object {
                "backgroundColors": Object {
                  "button": "#3fbaca",
                  "buttonDisabled": "#e2e2e2",
                  "common": "#ffffff",
                  "main": "#f7f7f7",
                  "secondaryToggle": "#ffffff",
                  "switchDisabled": "#e7e7e7",
                  "switchHandle": "#ffffff",
                  "switchOff": "#9d9d9d",
                  "switchOn": "#3fbaca",
                  "tag": "#293346",
                  "toggle": "#425579",
                },
                "borderColors": Object {
                  "common": "#e2e2e2",
                },
                "fontColors": Object {
                  "anchor": "#3fbaca",
                  "button": "#ffffff",
                  "common": "#293346",
                  "description": "#9d9d9d",
                  "disabledAnchor": "#9d9d9d",
                  "label": "#2d2d2d",
                  "secondaryToggle": "#9d9d9d",
                  "tableHeader": "#425579",
                  "tag": "#ffffff",
                  "text": "#2d2d2d",
                  "toggle": "#ffffff",
                },
              }
            }
            style={Object {}}
          >
            <div
              className="sc-bdnylx hdbdkm sc-gtssRu dZcusW"
              style={Object {}}
            >
              Description here!
            </div>
          </StyledBox>
        </Component>
      </Box>
    </StyledDescription>
  </Component>
</Description>
`;
